drivers/system_pathways/app/models/system_pathways/time_chart/ethnicity.rb

```diff
@@ -0,0 +1,156 @@
+###
+
+module SystemPathways::TimeChart::Ethnicity
+  extend ActiveSupport::Concern
+
+  private def ethnicity_chart_data
+    {
+      chart: 'ethnicity',
+      config: {
+        size: {
+          height: 2400,
+        },
+      },
+      data: ethnicity_data,
+      table: as_table(ethnicity_table_data, ['Project Type'] + ethnicities.values),
+      link_params: {
+        columns: [[]] + ethnicities.keys.map { |k| ['details[ethnicities][]', k] },
+        rows: [[]] + detail_node_keys.map { |k| ['node', k] },
+      },
+    }
+  end
+
+  private def ethnicity_table_data
+    race_counts[:ph_counts].transform_keys { |k| ph_projects[k] }
+    flat_counts = ethnicity_counts[:project_type_counts].merge(ethnicity_counts[:ph_counts])
+    flat_counts['Returned to Homelessness'] = ethnicity_counts[:return_counts]
+    flat_counts
+  end
+
+  private def ethnicity_counts # rubocop:disable Metrics/AbcSize
+    hispanic_latinaeo_column = sp_c_t['hispanic_latinaeo']
+    @ethnicity_counts ||= {}.tap do |e_counts|
+      project_type_counts = project_type_node_names.map do |label|
+        counts = {}
+        ethnicities.each_key do |k|
+          counts[k] ||= 0
+        end
+        # Get rid of the distinct from node_clients
+        stay_length_col = sp_e_t[:stay_length]
+        scope = SystemPathways::Enrollment.joins(:client).where(id: node_clients(label).select(:id))
+        race_data = pluck_to_hash(race_columns.except('race_none', 'multi_racial').map { |k, v| [sp_c_t[k], v] }.to_h.merge(stay_length_col => 'Stay Length'), scope)
+
+        data = race_data.select { |row| row[hispanic_latinaeo_column] }.map { |m| m[stay_length_col] }
+        counts[:hispanic_latinaeo] = average(data.sum, data.count).round
+        data = race_data.
+          select { |row| row.except(stay_length_col, hispanic_latinaeo_column).values.any?(true) }.
+          reject { |row| row[hispanic_latinaeo_column] }.
+          map { |m| m[stay_length_col] }
+        counts[:non_hispanic_latinaeo] = average(data.sum, data.count).round
+        data = race_data.select { |row| row.except(stay_length_col).values.all?(false) }.map { |m| m[stay_length_col] }
+        counts[:unknown] = average(data.sum, data.count).round
+        [
+          label,
+          counts,
+        ]
+      end.to_h
+      e_counts[:project_type_counts] = project_type_counts
+
+      ph_counts = ph_projects.map do |p_type, p_label|
+        counts = {}
+        ethnicities.each_key do |k|
+          counts[k] ||= 0
+        end
+        # Get rid of the distinct from node_clients
+        stay_length_col = sp_e_t[:days_to_move_in]
+        scope = SystemPathways::Enrollment.joins(:client).where(id: node_clients(p_type).select(:id)).where(stay_length_col.not_eq(nil))
+        race_data = pluck_to_hash(race_columns.except('race_none', 'multi_racial').map { |k, v| [sp_c_t[k], v] }.to_h.merge(stay_length_col => 'Days to Move-In'), scope)
```

Comment:
maybe someday we should add a method `known_individual_race_column` so we don't have to keep except-ing `race_none` and `multi_racial`, but probably ok for now.